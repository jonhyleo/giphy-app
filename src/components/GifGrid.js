import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <section>
        {loading ? (
          <div className="sk-chase">
            <div className="sk-chase-dot"></div>
            <div className="sk-chase-dot"></div>
            <div className="sk-chase-dot"></div>
            <div className="sk-chase-dot"></div>
            <div className="sk-chase-dot"></div>
            <div className="sk-chase-dot"></div>
          </div>
        ) : (
          <div className="row">
            <h3>{category}</h3>
            {images.map((img) => (
              <GifGridItem key={img.id} {...img} />
            ))}
          </div>
        )}
      </section>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
