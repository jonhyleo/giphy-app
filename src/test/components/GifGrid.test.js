import React from "react";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Tests cases for GifGrid component", () => {
  const category = "Dragon Ball";

  test("should render", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should show items when images are loaded using useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/image.jpg",
        title: "Title",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });
});
