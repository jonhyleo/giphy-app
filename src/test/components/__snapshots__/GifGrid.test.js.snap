// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests cases for GifGrid component should render 1`] = `
<Fragment>
  <section>
    <div
      className="sk-chase"
    >
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
    </div>
  </section>
</Fragment>
`;

exports[`Tests cases for GifGrid component should show items when images are loaded using useFetchGifs 1`] = `
<Fragment>
  <section>
    <div
      className="sk-chase"
    >
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
    </div>
  </section>
</Fragment>
`;
